nncf/openvino/quantization/weights_compression.py

```diff
@@ -0,0 +1,137 @@
+from nncf.openvino.graph.metatypes.openvino_metatypes import get_operation_const_op
+from nncf.openvino.graph.model_transformer import OVModelTransformer
+from nncf.openvino.graph.nncf_graph_builder import GraphConverter
+from nncf.openvino.graph.node_utils import get_const_value
+from nncf.openvino.statistics.statistics import OVMinMaxTensorStatistic
+from nncf.quantization.fake_quantize import FakeQuantizeParameters
+from nncf.quantization.fake_quantize import calculate_quantizer_parameters
+
+
+def insert_pre_compression_operations(model: ov.Model, bits: int = 8):
+    """
+    Inserts in-place weights compression with FakeQuantize operation for Linear and Embedding layers.
+
+    :param model: The original model to insert the weights compression.
+    :param bits: Number of bits for quantization.
+    """
+    allowed_metatypes_to_const_port = {OVEmbeddingMetatype: [0], OVMatMulMetatype: [0, 1]}
+    quantizer_config = QuantizerConfig(
+        num_bits=bits,
+        mode=QuantizationMode.ASYMMETRIC,
+        signedness_to_force=None,
+        per_channel=True,
+    )
+
+    get_fq_params = partial(
+        calculate_quantizer_parameters,
+        quantizer_config=quantizer_config,
+        quant_group=QuantizerGroup.WEIGHTS,
+        narrow_range=False,
+        half_range=False,
+    )
+
+    for node in model.get_ops():
+        # pylint:disable=protected-access
+        metatype = GraphConverter._get_node_metatype(node)
+        if metatype not in allowed_metatypes_to_const_port:
+            continue
+
+        for const_port_id in allowed_metatypes_to_const_port[metatype]:
+            weight_node = get_operation_const_op(node, const_port_id)
+            if weight_node is None:
+                continue
+
+            weight_output = weight_node.output(0)
+            fq_count = 0
+            for target_input in weight_output.get_target_inputs():
+                consumer_node = target_input.get_node()
+                if consumer_node.get_type_name() == "FakeQuantize":
+                    fq_count += 1
+
+            if fq_count > 0:
+                # FQ must be linked with all target inputs
+                assert fq_count == len(weight_output.get_target_inputs())
+                continue
+
+            weight = get_const_value(weight_node)
+            axes = _get_reduction_axes(metatype, node, const_port_id)
+            input_low = np.min(weight, axis=axes, keepdims=True)
+            input_high = np.max(weight, axis=axes, keepdims=True)
+            stats = OVMinMaxTensorStatistic(input_low, input_high)
+            fq_params = get_fq_params(stats)
+
+            node_name = node.get_friendly_name()
+            fq_name = f"{node_name}/fq_weights_{const_port_id}"
+            _insert_fake_quantize(fq_params, weight_output, fq_name)
+
+
+def _get_reduction_axes(metatype: Type[OperatorMetatype], node: ov.Node, const_port_id: int) -> Union[int, Tuple[int]]:
```

Comment:
Don't we need this method in the `node_utils.py` to reuse in the `MinMaxQuantization` backend? Or is it not the same?

Reply:
In `MinMaxQuantization` we use `NNCFGraph`. 
We don't create `NNCFGraph` from OpenVINO Model for `WeightCompression` to speed up algorithm.
I added `get_matmul_channel_axes` to `node_utils.py`